test/Transactions/Orleans.Transactions.Azure.Test/AzureTransactionLogTests.cs

```diff
@@ -30,35 +31,41 @@ public AzureTransactionLogTests()
         {
             var azureOptions = new AzureTransactionLogOptions()
             {
-                ArchiveLog = true,
                 ConnectionString = TestDefaultConfiguration.DataConnectionString,
                 TableName = TableName
             };
+            var archiveOptions = new AzureTransactionArchiveLogOptions()
+            {
+                ArchiveLog = true
+            };
 
-            var logStorage = await StorageFactory(azureOptions);
+            var logStorage = await StorageFactory(azureOptions, archiveOptions);
             var recordsNum = 2000;
             var allTransactions = new List<CommitRecord>(recordsNum);
             for (int i = 0; i< recordsNum; i++)
             {
                 allTransactions.Add(new CommitRecord(){LSN=i, TransactionId = i});
             }
             await logStorage.Append(allTransactions);
-            //a subset of all transactions will be archived 
-            await logStorage.TruncateLog(recordsNum + 2);
+            //all transactions will be archived 
+            var maxSLN = recordsNum + 2;
+            await logStorage.TruncateLog(maxSLN);
+            //since lsn and transaction id are the same in this test, so use it as the query on row key
             var query = new TableQuery<AzureTransactionLogStorage.ArchivalRow>().Where(
-                TableQuery.GenerateFilterCondition(nameof(AzureTransactionLogStorage.ArchivalRow.ClusterId), QueryComparisons.Equal, TestClusterId));
+                TableQuery.GenerateFilterCondition(nameof(AzureTransactionLogStorage.ArchivalRow.RowKey), QueryComparisons.LessThanOrEqual, AzureTransactionLogStorage.ArchivalRow.MakeRowKey(maxSLN)));
```

Comment:
Shouldn't query use minrowkey/maxrow key and limit search by partition?

Reply:
Using the entire query range ensures that archiving is not writing entities beyond where it should.

Limiting query to partition key prevents a table scan, which could be very expensive, especially for ci or other checks which may accumulate entities in the table over time.

Partition key also ensures that query is only querying entities from this deployment, in the case that multiple tests are being run at the same time using multiple serviceIds.

Probably should not delete table after run, this is not standard in our storage tests because it can cause failures in other test that may be running in parallel.  Also would hide bugs caused by improper partitioning of data.